components/playlist/browser/playlist_service.cc

```diff
@@ -1348,9 +1340,18 @@ base::SequencedTaskRunner* PlaylistService::GetTaskRunner() {
   return task_runner_.get();
 }
 
-void PlaylistService::RequestStreamingQuery(const std::string& query_id,
-                                            const std::string& url,
-                                            const std::string& method) {
+void PlaylistService::RequestStreamingQuery(
+    const std::string& query_id,
+    const std::string& url,
+    const std::string& method,
+    mojo::PendingRemote<mojom::PlaylistStreamingObserver> streaming_observer) {
+  if (streaming_observer_.is_bound()) {
+    streaming_observer_.reset();
+  }
+
+  streaming_observer_.Bind(std::move(streaming_observer));
+  mojo::Remote<mojom::PlaylistStreamingObserver> observer;
+  observer.Bind(std::move(streaming_observer));
```

Comment:
This doesn't seem to be correct. We are moving the stream observer twice `std::move(streaming_observer)`.

https://github.com/brave/brave-browser/issues/42229